commands/project/profile/add.ts

```diff
@@ -189,17 +188,42 @@ async function handler(
     );
 
     if (promptResponse) {
-      targetAccount = promptResponse;
+      targetAccountId = promptResponse;
+    }
+  }
+
+  const existingProfiles = await getAllHsProfiles(projectSourceDir);
+  let existingTargetAccountIdSelected = false;
+
+  for (const profile of existingProfiles) {
+    try {
+      const loadedProfile = loadHsProfileFile(projectSourceDir, profile);
+      if (loadedProfile?.accountId === targetAccountId) {
+        existingTargetAccountIdSelected = true;
+        break;
+      }
+    } catch (err) {
+      // Skip profiles that can't be loaded
+      continue;
```

Comment:
Nit: The `continue;` is redundant if it's at the end of the block.  Maybe this could be a debug log instead.